tests/api/test_store.py

```diff
@@ -390,3 +393,131 @@ async def test_api_store_addons_changelog_corrupted(
     assert resp.status == 200
     result = await resp.text()
     assert result == "Text with an invalid UTF-8 char: �"
+
+
+async def test_api_store_addons_addon_availability_success(
+    api_client: TestClient, store_addon: AddonStore
+):
+    """Test /store/addons/{addon}/availability REST API - success case."""
+    resp = await api_client.get(f"/store/addons/{store_addon.slug}/availability")
+    assert resp.status == 200
+
+
+async def test_api_store_addons_addon_availability_arch_not_supported(
+    api_client: TestClient, coresys: CoreSys
+):
+    """Test /store/addons/{addon}/availability REST API - architecture not supported."""
+    # Create an addon with unsupported architecture
+    addon_obj = AddonStore(coresys, "test_arch_addon")
+    coresys.addons.store[addon_obj.slug] = addon_obj
+
+    # Set addon config with unsupported architecture
+    addon_config = {
+        "advanced": False,
+        "arch": ["i386"],  # Not supported on current system
+        "slug": "test_arch",
+        "description": "Test arch add-on",
+        "name": "Test Arch Add-on",
+        "repository": "test",
+        "stage": "stable",
+        "version": "1.0.0",
+    }
+    coresys.store.data.addons[addon_obj.slug] = addon_config
+
+    # Mock the system architecture to be different
+    with patch.object(CpuArch, "supported", new=PropertyMock(return_value=["amd64"])):
+        resp = await api_client.get(f"/store/addons/{addon_obj.slug}/availability")
+        assert resp.status == 400
```

Comment:
Hm, since this is really an API to check, is `Bad Request` the right HTTP status? :thinking: